include/alpaka/core/ThreadPool.hpp

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2022 Antonio Di Pilato
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <atomic>
+#include <condition_variable>
+#include <functional>
+#include <future>
+#include <mutex>
+#include <queue>
+#include <thread>
+#include <vector>
+
+namespace alpaka::core
+{
+    class ThreadPool
+    {
+        using taskType = std::packaged_task<void()>;
```

Comment:
I'm not sure if there are strong naming conventions in alpaka... looking through the headers, _most_ of the types and type aliases are upper case, so maybe
```suggestion
        using Task = std::packaged_task<void()>;
```
?